clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -71,6 +71,7 @@ public class RequestBasedMetadata extends AbstractStoreMetadata {
   private final ClusterStats clusterStats;
   private final FastClientStats clientStats;
   private volatile boolean isServiceDiscovered;
+  private boolean isReady = false;
```

Comment:
We should make isReady volatile, otherwise we might end up with requests getting rejected even after metadata is initialized properly since the thread calling `start()/refresh()` will be different from the ones calling `isReady()`. See: https://www.baeldung.com/java-volatile